tsdb/head.go

```diff
@@ -258,26 +259,30 @@ func (h *Head) PostingsCardinalityStats(statsByLabelName string) *index.Postings
 // stripeSize sets the number of entries in the hash map, it must be a power of 2.
 // A larger stripeSize will allocate more memory up-front, but will increase performance when handling a large number of series.
 // A smaller stripeSize reduces the memory allocated, but can decrease performance with large number of series.
-func NewHead(r prometheus.Registerer, l log.Logger, wal *wal.WAL, chunkRange int64, stripeSize int) (*Head, error) {
+func NewHead(r prometheus.Registerer, l log.Logger, wal *wal.WAL, chunkRange int64, stripeSize int, seriesLifecycleCallback SeriesLifecycleCallback) (*Head, error) {
```

Comment:
```suggestion
func NewHead(r prometheus.Registerer, l log.Logger, wal *wal.WAL, chunkRange int64, stripeSize int, callback SeriesLifecycleCallback) (*Head, error) {
```

Reply:
let's maybe keep it simpler we don't have many callbacks (:

Reply:
But keeping it plain `callback` is confusing too, how about `seriesCallback`?